File: csharp-todo/SimpleTodoAPI/Models/Todo.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleTodoAPI.Models
{
    public class Todo
    {
        public int Id { get; set; }
        
        [Required]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;
        
        public bool Completed { get; set; } = false;
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public class CreateTodoRequest
    {
        [Required]
        public string Title { get; set; } = string.Empty;
    }

    public class UpdateTodoRequest
    {
        public string? Title { get; set; }
        public bool? Completed { get; set; }
    }
}

File: go-todo/models/todo.go
package models

import (
	"errors"
	"time"
)

// Todo はタスクを表現する構造体
type Todo struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	Title     string    `json:"title" gorm:"not null;size:200"`
	Completed bool      `json:"completed" gorm:"default:false"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// TableName はテーブル名を明示的に指定
func (Todo) TableName() string {
	return "todos"
}

// Validate はTodoのバリデーションを行う
func (t *Todo) Validate() error {
	if t.Title == "" {
		return errors.New("title is required")
	}
	if len(t.Title) > 200 {
		return errors.New("title is too long (max 200 characters)")
	}
	return nil
}
